src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -781,6 +785,57 @@ public boolean isAvailable()
                 getJDA().getEntityBuilder().createMember(this, resp.getObject()));
     }
 
+    @Nonnull
+    @Override
+    public CompletableFuture<MemberPresence> retrieveMemberPresence(long memberId)
+    {
+        Member member = getMemberById(memberId);
+        if (member != null
+            && api.isCacheFlagSet(CacheFlag.ACTIVITY)
+            && api.isCacheFlagSet(CacheFlag.CLIENT_STATUS))
+            return CompletableFuture.completedFuture(new MemberPresenceImpl(member));
+
+        CompletableFuture<MemberPresence> future = new CompletableFuture<>();
+        if (memberId == 0)
+        {
+            future.completeExceptionally(new IllegalArgumentException("Member with the specified userId does not exist"));
+            return future;
+        }
+
+        api.getClient().requestMembers((response) -> {
+            boolean notFound = response.optArray("not_found").map((i) -> i.length() > 1).orElse(false);
+            if (notFound)
+            {
+                future.completeExceptionally(new IllegalArgumentException("Member with the specified userId does not exist"));
+                return;
+            }
+
+            DataArray presences = response.getArray("presences");
+            if (presences.isEmpty())
+            {
+                future.complete(MemberPresenceImpl.EMPTY);
```

Comment:
What does it mean if we get no presences back but the user didn't result in 'not found'?

Reply:
It means the response was either not meant for this request or discord fucked up somehow.